File: week1/01-two-sum/260306.py
class Solution:
    def twoSum(self, nums, target):
        memo = {}
        answer = []

        for index, num in enumerate(nums):
            if target - num in memo:
                answer.append(memo[target - num])
                answer.append(index)
                break
            else:
                memo[num] = index

        return answer


if __name__ == '__main__':
    solution = Solution()
    print(solution.twoSum([2, 7, 11, 15], 9))

File: week1/01-two-sum/260306.js
var twoSum = function (nums, target) {
  const memo = {};
  const answer = [];

  nums.forEach((num, index) => {
    if (memo[target - num] === undefined) {
      memo[num] = index;
    } else {
      answer.push(memo[target - num]);
      answer.push(index);
    }
  });

  return answer;
};
